custom_components/sat/coordinator.py

```
@@ -423,74 +329,46 @@ async def async_added_to_hass(self) -> None:
 
     async def async_will_remove_from_hass(self) -> None:
         """Run when an entity is removed from hass."""
-        pass
+        await self._boiler.async_save_options()
 
-    async def async_control_heating_loop(self, climate: Optional[SatClimate] = None, pwm_state: Optional[PWMState] = None, _time=None) -> None:
+    async def async_control_heating_loop(self, climate: Optional[SatClimate] = None, pwm_status: Optional[PWMStatus] = None, timestamp: float = None) -> None:
```

Ruff (0.14.8)

Unused method argument: `climate`

(ARG002)